test/e2e/util/util.go

```diff
@@ -1167,6 +1195,434 @@ func EnsureNetworkPolicies(t *testing.T, ctx context.Context, c crclient.Client,
 	})
 }
 
+// InstallCilium validates that Cilium network policies are properly enforced
+// in ARO HCP guest clusters. This test covers:Verifying Cilium installation
+func InstallCilium(t *testing.T, ctx context.Context, guestClient crclient.Client, hostedCluster *hyperv1.HostedCluster) {
+	t.Run("InstallCilium", func(t *testing.T) {
+		if !azureutil.IsAroHCP() {
+			t.Skip("test only supported on ARO HCP clusters")
+		}
+
+		t.Run("InstallCiliumNameSpace", func(t *testing.T) {
+			g := NewWithT(t)
+			t.Log("Configuring cilium namespace with PodSecurity labels")
+			ciliumNs := &corev1.Namespace{}
+			err := guestClient.Get(ctx, crclient.ObjectKey{Name: "cilium"}, ciliumNs)
+
+			if apierrors.IsNotFound(err) {
+				// Namespace doesn't exist, create it with labels
+				t.Log("Creating cilium namespace")
+				ciliumNs = &corev1.Namespace{}
+				ciliumNs.Name = "cilium"
+				ciliumNs.Labels = map[string]string{
+					"security.openshift.io/scc.podSecurityLabelSync": "false",
+					"pod-security.kubernetes.io/enforce":             "privileged",
+					"pod-security.kubernetes.io/audit":               "privileged",
+					"pod-security.kubernetes.io/warn":                "privileged",
+				}
+				err = guestClient.Create(ctx, ciliumNs)
+				g.Expect(err).ToNot(HaveOccurred(), "failed to create cilium namespace")
+			} else {
+				// Namespace exists, update labels
+				g.Expect(err).ToNot(HaveOccurred(), "unexpected error checking cilium namespace")
+
+				t.Log("Updating existing cilium namespace with PodSecurity labels")
+				if ciliumNs.Labels == nil {
+					ciliumNs.Labels = make(map[string]string)
+				}
+				ciliumNs.Labels["security.openshift.io/scc.podSecurityLabelSync"] = "false"
+				ciliumNs.Labels["pod-security.kubernetes.io/enforce"] = "privileged"
+				ciliumNs.Labels["pod-security.kubernetes.io/audit"] = "privileged"
+				ciliumNs.Labels["pod-security.kubernetes.io/warn"] = "privileged"
```

Comment:
Just grab the manifest function once created and apply the content if that's not equal.